lib/ASTStmt.cpp

```diff
@@ -470,15 +485,141 @@ namespace pillar
+      case vast::hl::Predicate::ne:
+        return CreateNE(lhs_expr, rhs_expr);
+      case vast::hl::Predicate::slt:
+      case vast::hl::Predicate::ult:
+        return CreateLT(lhs_expr, rhs_expr);
+      case vast::hl::Predicate::sgt:
+      case vast::hl::Predicate::ugt:
+        return CreateGT(lhs_expr, rhs_expr);
+      case vast::hl::Predicate::sle:
+      case vast::hl::Predicate::ule:
+        return CreateLE(lhs_expr, rhs_expr);
+      case vast::hl::Predicate::sge:
+      case vast::hl::Predicate::uge:
+        return CreateGE(lhs_expr, rhs_expr);
+      default:
+        throw std::invalid_argument("Unsupported comparison operation");
+      }
+      return nullptr;
+    }
+    clang::ForStmt *AST::LiftForOp(clang::DeclContext *dc, mlir::Operation &op_)
+    {
+
+      vast::hl::ForOp op = mlir::dyn_cast<vast::hl::ForOp>(op_);
+      clang::Expr *cond_expr = nullptr;
+      for (mlir::Block &block : op.getCondRegion().getBlocks())
+      {
+        cond_expr = LiftBlockExpr(dc, block);
+        break;
+      }
+
+      clang::Expr *inc = nullptr;
+      for (mlir::Block &block : op.getIncrRegion())
+      {
+
+        inc = LiftBlockExpr(dc, block);
+      }
+      clang::CompoundStmt *body = LiftRegion(dc, op.getBodyRegion());
+
+      return CreateFor(nullptr, cond_expr, inc, body);
+    }
+    clang::UnaryOperator *AST::LiftPostIncOp(clang::DeclContext *dc, mlir::Operation &op_)
+    {
+      vast::hl::PostIncOp op = mlir::dyn_cast<vast::hl::PostIncOp>(op_);
+
+      clang::Expr *sub_expr = LiftValue(dc, op.getArg());
+      assert(sub_expr != nullptr);
+      return CreateUnaryOp(clang::UO_PostInc, sub_expr);
+    }
+    clang::UnaryOperator *AST::LiftPostDecOp(clang::DeclContext *dc, mlir::Operation &op_)
+    {
+      vast::hl::PostDecOp op = mlir::dyn_cast<vast::hl::PostDecOp>(op_);
+
+      clang::Expr *sub_expr = LiftValue(dc, op.getArg());
+      assert(sub_expr != nullptr);
+      return CreateUnaryOp(clang::UO_PostDec, sub_expr);
+    }
+    clang::UnaryOperator *AST::LiftPreIncOp(clang::DeclContext *dc, mlir::Operation &op_)
+    {
+      vast::hl::PreIncOp op = mlir::dyn_cast<vast::hl::PreIncOp>(op_);
+
+      clang::Expr *sub_expr = LiftValue(dc, op.getArg());
+      assert(sub_expr != nullptr);
+      return CreateUnaryOp(clang::UO_PreInc, sub_expr);
+    }
+    clang::UnaryOperator *AST::LiftPreDecOp(clang::DeclContext *dc, mlir::Operation &op_)
+    {
+      vast::hl::PreDecOp op = mlir::dyn_cast<vast::hl::PreDecOp>(op_);
+
+      clang::Expr *sub_expr = LiftValue(dc, op.getArg());
+      assert(sub_expr != nullptr);
+      return CreateUnaryOp(clang::UO_PreDec, sub_expr);
+    }
 
     // TODO(bmt): add more cases
     clang::Stmt *AST::LiftOpImpl(clang::DeclContext *dc, mlir::Operation &op)
     {
       switch (KindOf(op))
       {
       case HlOpKind::kUnknown:
-        assert(false);
-        return nullptr;
+        clang::Stmt *ret;
+        llvm::TypeSwitch<mlir::Operation *>(&op)
```

Comment:
`ret = nullptr;`

But really, you can use the second template operand to `TypeSwitch`, as follows:

```c++
llvm::TypeSwitch<mlir::Operation *, clang::Stmt *>(&op)
...
```

Then you can directly `return` the statements from your lambdas. The default case might need a `) -> clang::Stmt * {` as `nullptr` is technically a `std::nullptr_t`.